vmm/src/device_manager.rs

```diff
@@ -1472,22 +1670,79 @@ impl DeviceManager {
             id
         };
 
+        let mut node = device_node!(id);
+
+        // Look for the id in the device tree. If it can be found, that means
+        // the device is being restored, otherwise it's created from scratch.
+        let cache_range = if let Some(node) = self.device_tree.get(&id) {
+            debug!("Restoring virtio-fs {} resources", id);
+
+            let mut cache_range: Option<(u64, u64)> = None;
+            for resource in node.resources.iter() {
+                match resource {
+                    Resource::MmioAddressRange { base, size } => {
+                        if cache_range.is_some() {
+                            return Err(DeviceManagerError::ResourceAlreadyExist);
+                        }
+
+                        cache_range = Some((*base, *size));
+                    }
+                    _ => {
+                        error!("Unexpected resource for {}", id);
```

Comment:
Might as well print out the bad resource too.

Reply:
Done